### PR TITLE
Add generalized OCX API mocking/testing framework

### PR DESCRIPTION
**Pull request by jairideout**
## Status
- [x] **Ready**

## Description
Added a generalized API mocking/testing framework for writing unit tests against mocked One Codex APIs. This supports testing the `v1_experimental` API and future API versions.

Arbitrary One Codex API routes can be mocked by mirroring the API structure in `tests/data/api/`. More complex/custom requests can still be manually mocked in `conftest.py`.

Added dev docs to `README.md` describing how it all works. See the [rendered version](https://github.com/onecodex/onecodex/blob/d4d3f433b8cefff0e521591bdfad133decd772bd/README.md#writing-unit-tests).

Removed a bunch of unused data files and code from the previous testing framework.

Closes DEV-6947
Closes DEV-6987

## Related PRs
- [x] This PR is independent

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,6 +197,72 @@ make test
...
+│   ├── ...
+│   ├── schema
+│   │   ├── index.json  # payload for accessing GET::api/v1/schema
+│   │   └── index_all.json  # payload for accessing GET::api/v1/schema?expand=all
+│   └── ...
+└── v1_experimental
+    └── ...
+```
+
+The directory structure mirrors the One Codex API. For example:
```
Comment (safay) [inline README.md:238]: Very nice

**Comment by safay** [comment]
@jairideout The documentation and new dir structure that mirrors API routes helps make things much clearer. Sorry I don't have more constructive criticism. I'm itching to try this out and add some data for testing, and it looks to me like this clears the path forward.

**Review by safay — APPROVED** [review]
See my comment above. Looks good to me, though you may want someone with more engineering experience to comment on the details. Thanks for improving this!

File: tests/conftest.py [hunk 2]
```diff
@@ -269,72 +207,95 @@ def update_metadata_callback(req):
...
+                SCHEMA_ROUTES[resource_uri] = json.load(open(os.path.join(resource_root, filename)))
+        else:
+            for dirpath, _, filenames in os.walk(resource_root):
+                for filename in filenames:
+                    if filename not in {"index.json", "index.json.gz"}:
+                        continue
+
+                    filepath = os.path.join(dirpath, filename)
+                    if filepath.endswith(".json.gz"):
+                        # converting bytes (returned by gzip) to string is necessary in python 3.4
```
Comment (Keats) [inline tests/conftest.py:281]: Do we support python3.4? It's not even tested in the CI

**Review by Keats — APPROVED** [review]

**Review by petraszd — APPROVED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (jairideout) [reply tests/conftest.py]: 👍 removed, we support python 3.7+

**Pull request merged**